chore(self-managed): set index page for supported platforms

aabouzaid opened the pull request:
## What is the purpose of the change

Adding an index page for the `platforms` category where it has its own link as well as embedded auto-generated items. 

This will make it better for readability as well when we want to refer/share a specific category.

Here is the new introduced link (I will add it for 8.1 when the review is done).
https://docs.camunda.io/docs/next/self-managed/platform-deployment/helm-kubernetes/platforms/

And here is a screenshot for that new index page:
![c8sm-k8s-platforms](https://user-images.githubusercontent.com/6760103/201646460-94ee85f6-23e2-49f1-b999-95a21aad5289.png)


## Are there related marketing activities

No.

## When should this change go live?

No specific time, as soon as possible.

## PR Checklist

- [x] My changes apply to an already released version, and I have added them to the relevant `/versioned_docs` directory.
- [x] My changes apply to future versions, and I have added them to the main `/docs` directory.
- [x] My changes do not require an Engineering review.
- [x] My changes do not require a technical writer review.

aabouzaid commented on sidebars.js:
```diff
@@ -648,11 +648,19 @@ module.exports = {
+              label: "Platforms",
+              link: {
+                type: "doc",
+                id: "self-managed/platform-deployment/helm-kubernetes/platforms/platforms",
```
Regarding `platforms/platforms`, I'm using the same convention mentioned in the official docs for [Category index](https://docusaurus.io/docs/2.0.1/sidebar/autogenerated#category-index-convention).

pepopowitz reviewed (approved):
From a docs infrastructure perspective, I think this looks great! 

I defer to @christinaausley in regards to content & information architecture.

christinaausley commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/platforms.md:
```diff
@@ -0,0 +1,14 @@
+---
+id: platforms
+title: "Camunda Platform 8 officially supported Kubernetes platforms"
```
I'd ideally like to see a shorter title -- perhaps "Supported Kubernetes platforms"?

christinaausley commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/platforms.md:
```diff
@@ -0,0 +1,14 @@
+import DocCardList from '@theme/DocCardList';
+import {useCurrentSidebarCategory} from '@docusaurus/theme-common';
+
+[Deploying Camunda Platform 8 using Helm charts](../deploy.md) will work for all platforms and distributions. However, each platform or cloud provider may have some special prerequisites or pitfalls to avoid. The platforms guides highlights important notes for a smooth Camunda Platform deployment on various platforms.
```
"The platforms guides highlights important notes for a smooth Camunda Platform deployment on various platforms."

Do you mean "The Platform guides highlight important notes..."? Could we link to what you've mentioned here?

aabouzaid replied:
@christinaausley I've rephrased it. I was referencing to "Kubernetes platform" not the "Camunda Platform".

Regarding the links, they are auto-generated at the end of the page, as shown in this screenshot:
![201646460-94ee85f6-23e2-49f1-b999-95a21aad5289](https://user-images.githubusercontent.com/6760103/202098738-3fdc99b1-e90a-4367-afcf-aaaa84852ce4.png)

christinaausley reviewed (changes requested):
A few comments for clarification and some adjustments to the wording in my latest commit -- let me know if you have any questions about these.

aabouzaid commented:
@pepopowitz thanks for the review :+1: 

@christinaausley I've addressed your comments but by mistake, I overrode your "commit" :see_no_evil: 
so could you please push it again? :bow:

aabouzaid commented:
This PR is ready to be squashed and merged :heavy_check_mark: (github removed the previous approval)
